From a9dc50330acee2ac4fe4e09677ec71885e7baf08 Mon Sep 17 00:00:00 2001
From: Dmitry Shachnev <mitya57@gmail.com>
Date: Thu, 11 Apr 2013 17:32:00 +0400
Subject: [PATCH 001/132] QGtkStyle: remove an unnecessary sanity check for
 theme name

This makes QGtkStyle working again on Debian experimental
and Ubuntu 13.04, where getThemeName() returns empty value.

Backported from QtBase b72aa1cad30a44b1.

Change-Id: If4cc822970088a5fb31aa28d9951db1efc0bb4fd
Reviewed-by: J-P Nurmi <jpnurmi@digia.com>
---
 src/gui/styles/qgtkstyle_p.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/gui/styles/qgtkstyle_p.cpp b/src/gui/styles/qgtkstyle_p.cpp
index a21eead..302cf09 100644
--- a/src/gui/styles/qgtkstyle_p.cpp
+++ b/src/gui/styles/qgtkstyle_p.cpp
@@ -507,10 +507,7 @@ void QGtkStylePrivate::initGtkWidgets() const
     if (!gtkWidgetMap()->contains("GtkWindow") && themeName.isEmpty()) {
         themeName = getThemeName();
 
-        if (themeName.isEmpty()) {
-            qWarning("QGtkStyle was unable to detect the current GTK+ theme.");
-            return;
-        } else if (themeName == QLS("Qt") || themeName == QLS("Qt4")) {
+        if (themeName == QLS("Qt") || themeName == QLS("Qt4")) {
             // Due to namespace conflicts with Qt3 and obvious recursion with Qt4,
             // we cannot support the GTK_Qt Gtk engine
             qWarning("QGtkStyle cannot be used together with the GTK_Qt engine.");
-- 
1.8.3.1

